environments/fetcher/fetcher.go

```diff
@@ -293,14 +294,16 @@ func (fetcher *Fetcher) Fetch(req FetchRequest) (int, error) {
 	}
 
 	// check file type here, if the file is a zip file unarchive it.
-	if archiver.Zip.Match(tmpPath) {
+	// If extract true then extract else use archive as it is
```

Comment:
Please improve comment, since the code doesnt just check if the extract flag is true, it also checks if it's an archive.  Though the code seems readable and I would be ok with just removing the comment